### PR TITLE
Use renameOrCopy for single-file move operation

### LINKED ISSUE #357: Cross-device 'single file' copying on windows fails
```
PS W:\> .\krew.exe install --manifest=krew.yaml
Installing plugin: krew
W1029 12:35:34.778118   28440 install.go:127] failed to install plugin "krew": install failed: failed while moving files to the installation directory: could not rename file from "C:\\Users\\h\\AppData\\Local\\Temp\\krew-temp-move316849743" to "W:\\.krew\\store\\krew\\v0.3.1": rename C:\Users\h\AppData\Local\Temp\krew-temp-move316849743 W:\.krew\store\krew\v0.3.1: The system cannot move the file to a different disk drive.
F1029 12:35:34.781121   28440 root.go:58] failed to install some plugins: [krew]
```

- Windows
- Following installation instructions
- Running the install command both from C: and W: fails

### PR DESCRIPTION
**Pull request by ahmetb**
It looks like we can generalize copyDir func as "copy" because filepath.Walk
works on regular files, too. Making single file move use this renameOrCopy
method.

Also fixed a typo.

This should fix #357.

### CODE REVIEW

**Comment by ahmetb** [comment]
I have not fully tested it.

I had to plug in an SD card on my windows machine to create a true different device (subst command wasn't doing that). And since SD card is FAT, it can't create symlinks.

But this has enabled it to progress installation further than before, and fail at the last step (which is creating the symlink).

**Comment by ahmetb** [comment]
Also got this feeling like move.go is unnecessarily complex. We could most likely simplify this file using more idiomatic methods from stdlib.

File: pkg/installation/move.go [hunk 1]
```diff
@@ -196,18 +196,19 @@ func moveOrCopyDir(from, to string) error {
 		glog.V(4).Infof("Move target directory %q cleaned up", to)
 	}
 
-	err = os.Rename(from, to)
+	err = copy(from, to)
```
Comment (corneliusweig) [inline pkg/installation/move.go]: This basically means, that we don't even try to rename files anymore, right? This is a bit sad. I wish there were a better way.

If I'm correct, then we don't need the rename-specific error handling below.

Comment (ahmetb) [reply pkg/installation/move.go]: Hmm wait this is weird. I think we need to keep this as `os.Rename`, so we run into the LinkError below. I'll update, but likely can't test right now.

Comment (ahmetb) [reply pkg/installation/move.go]: Fixed.

**Comment by ferhatelmas** [comment]
> Also got this feeling like move.go is unnecessarily complex. We could most likely simplify this file using more idiomatic methods from stdlib.

I will give a look if I can simplify anything. Unless possible via stdlib, are we open to a dependency that does it?

File: pkg/installation/move.go [hunk 2]
```diff
@@ -201,13 +201,14 @@ func moveOrCopyDir(from, to string) error {
 	if le, ok := err.(*os.LinkError); err != nil && ok {
 		if errno, ok := le.Err.(syscall.Errno); ok && errno == 18 {
 			glog.V(4).Infof("Cross-device link error (ERRNO=18), fallback to manual copy")
-			return copyDir(from, to)
+			return copy(from, to)
```
Comment (ferhatelmas) [inline pkg/installation/move.go]: What is our stand overriding built-ins?

File: pkg/installation/move.go [hunk 3]
```diff
@@ -201,13 +201,14 @@ func moveOrCopyDir(from, to string) error {
...
 		}
 	}
 	return err
 }
 
-func copyDir(from string, to string) (err error) {
+// copy copies files or directories, recursively.
+func copy(from string, to string) (err error) {
```
Comment (ferhatelmas) [inline pkg/installation/move.go]: first type could be dropped: `from, to string`

File: pkg/installation/move.go (hunk 2 above)
Comment (corneliusweig) [reply pkg/installation/move.go]: I would like to avoid that.

Comment (ahmetb) [reply pkg/installation/move.go]: Go is surprisingly forgiving when it comes to these things.

I will rename this to `copyTree`.

**Comment by corneliusweig** [comment]
/lgtm
/approve
/hold
Because it needs to be verified again (or you remove the fix #357 from the PR description)

**Comment by ahmetb** [comment]
> I will give a look if I can simplify anything. Unless possible via stdlib, are we open to a dependency that does it? Also, should I create an issue to track it ?

I recommend looking at the move.go a bit.
The naming of the methods makes the hierarchy unclear:

- moveToInstallDir (extractDir &rarr; storeDir)
  - moveAllFiles (extractDir &rarr; tmp)
    - moveFiles
      - findMoveTargets
        - getDirectMove
          - isMoveAllowed
        - isMoveAllowed
      - renameOrCopy
  - renameOrCopy (tmp &rarr; storeDir)

For example, I'm not sure why is "getDirectMove" is a special case. There's already a filepath.Glob code in that method. Could that be already handling single file moves?

Maybe needs fixing, maybe not.

^ Feel free to copy/paste this into a new issue.

**Comment by corneliusweig** [comment]
@ferhatelmas also raised the question whether we should look out for an external dependency to do the moving. What's your optionion on this @ahmetb ?

**Comment by ahmetb** [comment]
I'm not aware of anything that can solve this out-of-the box, but I'm open to it. Our logic has some stuff like supporting globs (although `doublestar` pkg works better for that), checking if a move is allowed (e.g. `to: ../../../`).

**Comment by ahmetb** [comment]
I'll remove the fix myself I think. I don't have multiple real partitions on windows to test this but I think I can do

- plug in SD card to E: (FAT)
- set TMP=E:\tmp
- set KREW_ROOT=C:\users\ahmetb\.krew
- krew install
  - symlink to c:\

this would ensure we try to os.Rename files from e:\ to c:\, which would fall back to copying. At the end it would symlink to C:\ (NTFS) which is allowed.

**Comment by haf-afa** [comment]
My other directory is not a partition; it's a network attached file system that %HOME% is redirected to in corporate environments.

**Comment by corneliusweig** [comment]
@haf-afa Sounds like the perfect environment to test it ;) -- Provided this is possible for you.

**Comment by ahmetb** [comment]
@haf-afa is it possible you can test the fix please? Without your help, it's unlikely we'll be able to figure this out. I'm attaching the steps below.

I have a suspicion that symlinks won't work on a NAS filesystem where your %HOME% resides.

To test the fix in #375, do these steps:

1. download 
[krew.exe.zip](https://github.com/kubernetes-sigs/krew/files/3838913/krew.exe.zip) and extract it to get a krew.exe out
2. download [krew.yaml.txt](https://github.com/kubernetes-sigs/krew/files/3838914/krew.yaml.txt) and rename it to krew.yaml
3. download [krew.tar.gz](https://github.com/kubernetes-sigs/krew/files/3838916/krew.tar.gz)
4. run: `krew.exe --manifest=krew.yaml --archive=krew.tar.gz`
5. if it fails, delete `%HOME%\.krew` and re-run the command above with flag `-v=5` and copy its output here

**Comment by haf-afa** [comment]
```
PS W:\> .\krew.exe install --manifest=krew.yaml --archive=krew.tar.gz
Installing plugin: krew
W1113 11:01:12.382633   15724 install.go:127] failed to install plugin "krew": install failed: failed while moving files to the installation directory: could not rename file from "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move808672987" to "W:\\.krew\\store\\krew\\v0.3.2-4-g5cca27b": rename C:\Users\hefe01\AppData\Local\Temp\krew-temp-move808672987 W:\.krew\store\krew\v0.3.2-4-g5cca27b: The system cannot move the file to a different disk drive.
F1113 11:01:12.384635   15724 root.go:58] failed to install some plugins: [krew]
PS W:\> .\krew.exe install --manifest=krew.yaml --archive=krew.tar.gz -v=5
I1113 11:01:54.560546   16504 root.go:98] detected windows, will check for old krew installations to clean up
I1113 11:01:54.562067   16504 root.go:111] could not find krew's own plugin receipt, skipping cleanup of stale krew installations
I1113 11:01:54.562593   16504 install.go:144] --manifest specified, not ensuring plugin index
I1113 11:01:54.567626   16504 install.go:113] Will install plugin: krew
Installing plugin: krew
I1113 11:01:54.567626   16504 install.go:59] Looking for installed versions
I1113 11:01:54.568590   16504 platform.go:43] Matching platform for labels(arch=amd64,os=windows)
I1113 11:01:54.568590   16504 platform.go:51] Found matching platform with index (3)
I1113 11:01:54.569591   16504 install.go:78] Install plugin krew at version=v0.3.2-4-g5cca27b
I1113 11:01:54.570591   16504 install.go:96] Creating download staging directory "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew"
I1113 11:01:54.572593   16504 downloader.go:36] Fetching "https://github.com/kubernetes-sigs/krew/releases/download/v0.3.2-4-g5cca27b/krew.tar.gz"
I1113 11:01:54.577590   16504 downloader.go:43] Reading download data into memory
I1113 11:01:54.649589   16504 downloader.go:48] Read 4324320 bytes of download data into memory
I1113 11:01:54.650624   16504 verifier.go:51] Compare sha256 (6efc87a33df6da4441ad85cf2673c9741083b736849c7c6e48d53a02a7cded69) signed version
I1113 11:01:54.651623   16504 downloader.go:204] detected "application/x-gzip" file type
I1113 11:01:54.652588   16504 downloader.go:101] tar: extracting to "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew"
I1113 11:01:54.653590   16504 downloader.go:119] tar: processing "./krew-windows_amd64.exe" (type=48, mode=-rwxr-xr-x)
I1113 11:01:54.653590   16504 downloader.go:138] tar: ensuring parent dirs exist for regular file, dir=C:\Users\hefe01\AppData\Local\Temp\krew-downloads\krew
I1113 11:01:54.811613   16504 downloader.go:155] tar: processed "./krew-windows_amd64.exe"
I1113 11:01:54.811613   16504 downloader.go:157] tar extraction to C:\Users\hefe01\AppData\Local\Temp\krew-downloads\krew complete
I1113 11:01:54.813589   16504 move.go:156] Creating plugin installation directory "W:\\.krew\\store\\krew\\v0.3.2-4-g5cca27b"
I1113 11:01:54.822585   16504 move.go:162] Creating temp plugin move operations dir "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move322136223"
I1113 11:01:54.823587   16504 move.go:125] Finding move targets from "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew" to "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move322136223" with file operation=index.FileOperation{From:"./krew-windows_amd64.exe", To:"krew.exe"}
I1113 11:01:54.824589   16504 move.go:44] Trying to move single file directly from="C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew" to="C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move322136223" with file operation=index.FileOperation{From:"./krew-windows_amd64.exe", To:"krew.exe"}
I1113 11:01:54.825586   16504 move.go:48] Detected single move from file operation=index.FileOperation{From:"./krew-windows_amd64.exe", To:"krew.exe"}
I1113 11:01:54.825586   16504 move.go:132] Move file from "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew\\krew-windows_amd64.exe" to "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move322136223\\krew.exe"
I1113 11:01:54.827633   16504 move.go:141] Move operations are complete
I1113 11:01:54.827633   16504 move.go:172] Move directory "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move322136223" to "W:\\.krew\\store\\krew\\v0.3.2-4-g5cca27b"
I1113 11:01:54.828622   16504 move.go:192] There's already a directory at move target "W:\\.krew\\store\\krew\\v0.3.2-4-g5cca27b". deleting.
I1113 11:01:54.833764   16504 move.go:196] Move target directory "W:\\.krew\\store\\krew\\v0.3.2-4-g5cca27b" cleaned up
I1113 11:01:54.834585   16504 move.go:175] Cleaning up installation directory due to error during copying files
I1113 11:01:54.836586   16504 install.go:101] Deleting the download staging directory C:\Users\hefe01\AppData\Local\Temp\krew-downloads\krew
W1113 11:01:54.837588   16504 install.go:127] failed to install plugin "krew": install failed: failed while moving files to the installation directory: could not rename file from "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move322136223" to "W:\\.krew\\store\\krew\\v0.3.2-4-g5cca27b": rename C:\Users\hefe01\AppData\Local\Temp\krew-temp-move322136223 W:\.krew\store\krew\v0.3.2-4-g5cca27b: The system cannot move the file to a different disk drive.
F1113 11:01:54.838588   16504 root.go:56] failed to install some plugins: [krew]
sigs.k8s.io/krew/cmd/krew/cmd.init.1.func1
        /Users/ahmetb/workspace/krew/cmd/krew/cmd/install.go:138
github.com/spf13/cobra.(*Command).execute
        /Users/ahmetb/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:762
github.com/spf13/cobra.(*Command).ExecuteC
        /Users/ahmetb/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852
github.com/spf13/cobra.(*Command).Execute
        /Users/ahmetb/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
sigs.k8s.io/krew/cmd/krew/cmd.Execute
        /Users/ahmetb/workspace/krew/cmd/krew/cmd/root.go:54
main.main
        /Users/ahmetb/workspace/krew/cmd/krew/main.go:24
runtime.main
        /usr/local/Cellar/go/1.13.4/libexec/src/runtime/proc.go:203
runtime.goexit
        /usr/local/Cellar/go/1.13.4/libexec/src/runtime/asm_amd64.s:1357
```

**Comment by haf-afa** [comment]
To clarify; the above log happens **despite** W:\.krew NOT existing before I run the command. I also don't know why it downloads something when I have the archive just beside.

**Comment by ahmetb** [comment]
@haf-afa thanks for trying, it’s odd this is still happening I'll take another look.

It's not downloading a file (url doesn't exist anyway), but the logs are misleading (@corneliusweig I thought we fixed this?)

**Comment by ahmetb** [comment]
It seems like we're not correctly detecting the win32 error and executing the fallback

> glog.V(4).Infof("Cross-device link error (ERRNO=18), fallback to manual copy")

When I search the error message `The system cannot move the file to a different disk drive.` I see that that's a different error code (referred to as `17`, `0x11`, or `0x800700111`).

**Comment by ahmetb** [comment]
/cc @corneliusweig 

This is awkward, last week I saw here: https://docs.microsoft.com/en-us/cpp/c-runtime-library/errno-doserrno-sys-errlist-and-sys-nerr?view=vs-2019 that the EXDEV equivalent in windows has code `18` (0x12).

Then, on my windows machine I tried a cross-fs copy and it worked, further made me think 18 (0x12) is correct. But I didn't pay attention to whether if the log message showed up there. (I will repeat this exercise.)

Searching the original error message on the internet and looking this Go code makes me believe the windows equivalent of EXDEV is definitely `17` (0x11):

https://github.com/golang/sys/blob/4c7a9d0fe056d9d1de37e1409ca8a5c17accb46a/windows/zerrors_windows.go#L181

So it seems like we should factor `errcode==18` to:

    (IsWindows() && errcode==17) || (!IsWindows() && errcode==18)

**Comment by ahmetb** [comment]
@haf-afa I think we finally nailed this time. It turns out the equivalent of the error code on windows was off-by-one.
Can you try this ee1b291 build with these files:

[krew.yaml.txt](https://github.com/kubernetes-sigs/krew/files/3844419/krew.yaml.txt)
[krew.exe.zip](https://github.com/kubernetes-sigs/krew/files/3844429/krew.exe.zip)
[krew.tar.gz](https://github.com/kubernetes-sigs/krew/files/3844434/krew.tar.gz)

Same instructions [as above](#issuecomment-553189184), make sure to delete `%HOME%\.krew` first.

**Comment by haf-afa** [comment]
I appreciate you working on this. Thank you.

Next error; symlinks are so advanced for Windows users that they require admin privileges to use.

Symlinks


```
PS W:\> .\krew.exe install --manifest=krew.yaml --archive=krew.tar.gz -v=5
I1114 09:37:59.053043   14916 root.go:98] detected windows, will check for old krew installations to clean up
I1114 09:37:59.054066   14916 root.go:111] could not find krew's own plugin receipt, skipping cleanup of stale krew installations
I1114 09:37:59.055044   14916 install.go:144] --manifest specified, not ensuring plugin index
I1114 09:37:59.061045   14916 install.go:113] Will install plugin: krew
Installing plugin: krew
I1114 09:37:59.062047   14916 install.go:59] Looking for installed versions
I1114 09:37:59.062047   14916 platform.go:43] Matching platform for labels(arch=amd64,os=windows)
I1114 09:37:59.062047   14916 platform.go:51] Found matching platform with index (3)
I1114 09:37:59.063043   14916 install.go:78] Install plugin krew at version=v0.3.2-5-gee1b291
I1114 09:37:59.063043   14916 install.go:96] Creating download staging directory "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew"
I1114 09:37:59.064045   14916 downloader.go:36] Fetching "https://github.com/kubernetes-sigs/krew/releases/download/v0.3.2-5-gee1b291/krew.tar.gz"
I1114 09:37:59.067044   14916 downloader.go:43] Reading download data into memory
I1114 09:37:59.325145   14916 downloader.go:48] Read 4324849 bytes of download data into memory
I1114 09:37:59.325145   14916 verifier.go:51] Compare sha256 (8f9b083de4abf5a1fb0a812e61daf2b27a855dd2c934f56bf56ef578884859ee) signed version
I1114 09:37:59.326141   14916 downloader.go:204] detected "application/x-gzip" file type
I1114 09:37:59.326141   14916 downloader.go:101] tar: extracting to "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew"
I1114 09:37:59.327141   14916 downloader.go:119] tar: processing "./krew-windows_amd64.exe" (type=48, mode=-rwxr-xr-x)
I1114 09:37:59.327141   14916 downloader.go:138] tar: ensuring parent dirs exist for regular file, dir=C:\Users\hefe01\AppData\Local\Temp\krew-downloads\krew
I1114 09:37:59.512230   14916 downloader.go:155] tar: processed "./krew-windows_amd64.exe"
I1114 09:37:59.513230   14916 downloader.go:157] tar extraction to C:\Users\hefe01\AppData\Local\Temp\krew-downloads\krew complete
I1114 09:37:59.513230   14916 move.go:156] Creating plugin installation directory "W:\\.krew\\store\\krew\\v0.3.2-5-gee1b291"
I1114 09:37:59.530247   14916 move.go:162] Creating temp plugin move operations dir "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move318840203"
I1114 09:37:59.532238   14916 move.go:125] Finding move targets from "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew" to "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move318840203" with file operation=index.FileOperation{From:"./krew-windows_amd64.exe", To:"krew.exe"}
I1114 09:37:59.533231   14916 move.go:44] Trying to move single file directly from="C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew" to="C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move318840203" with file operation=index.FileOperation{From:"./krew-windows_amd64.exe", To:"krew.exe"}
I1114 09:37:59.534232   14916 move.go:48] Detected single move from file operation=index.FileOperation{From:"./krew-windows_amd64.exe", To:"krew.exe"}
I1114 09:37:59.534232   14916 move.go:132] Move file from "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-downloads\\krew\\krew-windows_amd64.exe" to "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move318840203\\krew.exe"
I1114 09:37:59.540232   14916 move.go:141] Move operations are complete
I1114 09:37:59.543231   14916 move.go:172] Move directory "C:\\Users\\hefe01\\AppData\\Local\\Temp\\krew-temp-move318840203" to "W:\\.krew\\store\\krew\\v0.3.2-5-gee1b291"
I1114 09:37:59.548231   14916 move.go:192] There's already a directory at move target "W:\\.krew\\store\\krew\\v0.3.2-5-gee1b291". deleting.
I1114 09:37:59.579758   14916 move.go:196] Move target directory "W:\\.krew\\store\\krew\\v0.3.2-5-gee1b291" cleaned up
I1114 09:37:59.580764   14916 move.go:202] Cross-device link error while copying, fallback to manual copy
I1114 09:37:59.581776   14916 move.go:216] Creating new dir "W:\\.krew\\store\\krew\\v0.3.2-5-gee1b291"
I1114 09:37:59.596777   14916 move.go:219] Copying file "W:\\.krew\\store\\krew\\v0.3.2-5-gee1b291\\krew.exe"
I1114 09:38:00.254834   14916 install.go:213] No file found at "W:\\.krew\\bin\\kubectl-krew.exe"
I1114 09:38:00.255688   14916 install.go:200] Creating symlink to "W:\\.krew\\store\\krew\\v0.3.2-5-gee1b291\\krew.exe" at "W:\\.krew\\bin\\kubectl-krew.exe"
I1114 09:38:00.262840   14916 install.go:101] Deleting the download staging directory C:\Users\hefe01\AppData\Local\Temp\krew-downloads\krew
W1114 09:38:00.263690   14916 install.go:127] failed to install plugin "krew": install failed: failed to link installed plugin: failed to create a symlink from "W:\\.krew\\store\\krew\\v0.3.2-5-gee1b291\\krew.exe" to "W:\\.krew\\bin\\kubectl-krew.exe": symlink W:\.krew\store\krew\v0.3.2-5-gee1b291\krew.exe W:\.krew\bin\kubectl-krew.exe: A required privilege is not held by the client.
F1114 09:38:00.264696   14916 root.go:56] failed to install some plugins: [krew]
sigs.k8s.io/krew/cmd/krew/cmd.init.1.func1
        /Users/ahmetb/workspace/krew/cmd/krew/cmd/install.go:138
github.com/spf13/cobra.(*Command).execute
        /Users/ahmetb/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:762
github.com/spf13/cobra.(*Command).ExecuteC
        /Users/ahmetb/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852
github.com/spf13/cobra.(*Command).Execute
        /Users/ahmetb/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
sigs.k8s.io/krew/cmd/krew/cmd.Execute
        /Users/ahmetb/workspace/krew/cmd/krew/cmd/root.go:54
main.main
        /Users/ahmetb/workspace/krew/cmd/krew/main.go:24
runtime.main
        /usr/local/Cellar/go/1.13.4/libexec/src/runtime/proc.go:203
runtime.goexit
        /usr/local/Cellar/go/1.13.4/libexec/src/runtime/asm_amd64.s:1357
```

**Comment by corneliusweig** [comment]
@haf-afa Lol, that made me laugh big time :rofl:. What you have discovered is a known limitation of `krew` (#378). For the time being, you need to install plugins with admin privileges :cry:.

Anyway, it looks like @ahmetb has finally found the correct way to handle this :champagne: :tada: :rocket:. Great work! And thanks @haf-afa for testing repeatedly.

File: pkg/installation/move.go [hunk 4]
```diff
@@ -243,3 +242,17 @@ func copyFile(source string, dst string, mode os.FileMode) (err error) {
...
+func isCrossDeviceRenameErr(err error) bool {
+	le, ok := err.(*os.LinkError)
+	if !ok {
+		return false
+	}
+	errno, ok := le.Err.(syscall.Errno)
+	if !ok {
+		return false
+	}
+	return (IsWindows() && errno == 17) || // syscall.ERROR_NOT_SAME_DEVICE
```
Comment (corneliusweig) [inline pkg/installation/move.go]: :+1: for putting the constant names in the comments

**Comment by corneliusweig** [comment]
Great work, thanks for investing so much effort to get to the root of the problem!
/lgtm
/approve

File: pkg/installation/move.go (hunk 4 above)
Comment (ferhatelmas) [reply pkg/installation/move.go]: btw, a search brought this link, kind of same implementation but better documentation

https://github.com/golang/dep/blob/master/internal/fs/rename_windows.go

**Comment by ahmetb** [comment]
@haf-afa Please try enabling Developer Mode on windows and try again. That should not require Administrator permissions.

**Pull request merged**